jerry-core/ecma/builtin-objects/ecma-builtin-helpers-sort.c

```diff
@@ -54,35 +58,39 @@ ecma_builtin_helper_array_to_heap (ecma_value_t *array_p, /**< heap data array *
         child++;
       }
 
-      ECMA_FINALIZE (child_compare_value);
+      ecma_free_value (child_compare_value);
     }
 
-    if (ecma_is_value_empty (ret_value))
-    {
-      JERRY_ASSERT (child <= right);
+    JERRY_ASSERT (child <= right);
 
-      /* Compare current child node with the swap (tree top). */
-      ECMA_TRY_CATCH (swap_compare_value, sort_cb (array_p[child], swap, compare_func), ret_value);
-      JERRY_ASSERT (ecma_is_value_number (swap_compare_value));
+    /* Compare current child node with the swap (tree top). */
+    ecma_value_t swap_compare_value = sort_cb (array_p[child], swap, compare_func);
 
-      if (ecma_get_number_from_value (swap_compare_value) <= ECMA_NUMBER_ZERO)
-      {
-        /* Break from loop if current child is less than swap (tree top) */
-        should_break = true;
-      }
-      else
-      {
-        /* We have to move 'swap' lower in the tree, so shift current child up in the hierarchy. */
-        uint32_t parent = (child - 1) / 2;
-        JERRY_ASSERT (parent <= right);
-        array_p[parent] = array_p[child];
+    if (ECMA_IS_VALUE_ERROR (swap_compare_value))
+    {
+      ret_value = ECMA_VALUE_ERROR;
+      break;
+    }
 
-        /* Update child to be the left child of the current node. */
-        child = child * 2 + 1;
-      }
+    JERRY_ASSERT (ecma_is_value_number (swap_compare_value));
+
+    if (ecma_get_number_from_value (swap_compare_value) <= ECMA_NUMBER_ZERO)
+    {
+      /* Break from loop if current child is less than swap (tree top) */
+      should_break = true;
```

Comment:
How about use `break` instead of this variable, just call `ecma_free_value (swap_compare_value)` before it.